readthedocs/rtd_tests/utils.py

```diff
@@ -55,34 +60,16 @@ def make_test_git():
     # URL are not allowed and using a real URL will require Internet to clone
     # the repo
     check_output(['git', 'checkout', '-b', 'submodule', 'master'], env=env)
-    # https://stackoverflow.com/a/37378302/2187091
-    mkdir(pjoin(directory, 'foobar'))
-    gitmodules_path = pjoin(directory, '.gitmodules')
-    with open(gitmodules_path, 'w') as fh:
-        fh.write('''[submodule "foobar"]\n\tpath = foobar\n\turl = https://foobar.com/git\n''')
-    check_output(
-        [
-            'git', 'update-index', '--add', '--cacheinfo', '160000',
-            '233febf4846d7a0aeb95b6c28962e06e21d13688', 'foobar',
-        ],
-        env=env,
+    add_submodule_without_cloning(
+        directory, 'foobar', 'https://foobar.com/git'
     )
     check_output(['git', 'add', '.'], env=env)
     check_output(['git', 'commit', '-m"Add submodule"'], env=env)
 
-    # Add a relative submodule URL in the relativesubmodule branch
-    check_output(['git', 'checkout', '-b', 'relativesubmodule', 'master'], env=env)
```

Comment:
This wasn't creating the submodule as expected, and we have `add_submodule_without_cloning` now.